assets/queries/cloudFormation/aws/default_security_groups_with_unrestricted_traffic/query.rego

```diff
@@ -1,35 +1,83 @@
+	search_values := check_rules(resource.Properties, security_group_name, input.document[i])
+	search_values != ""
 
 	result := {
 		"documentId": input.document[i].id,
-		"resourceType": resource.Type,
-		"resourceName": cf_lib.get_resource_name(resource, name),
-		"searchKey": sprintf("%s%s.Properties", [cf_lib.getPath(path), name]),
+		"resourceType": "AWS::EC2::SecurityGroup",
+		"resourceName": cf_lib.get_resource_name(resource, security_group_name),
+		"searchKey": search_values.searchKey,
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": sprintf("Resources.%s.Properties.GroupName should be defined as default and the inbound and outbound rules should be empty.", [name]),
-		"keyActualValue": sprintf("Resources.%s.Properties.GroupName is defined as default and the inbound and outbound rules are not empty.", [name]),
+		"keyExpectedValue": "Any 'AWS::EC2::SecurityGroup' with 'Properties.GroupName' set to 'default' should not have any traffic rules set.",
+		"keyActualValue": sprintf("'Resources.%s' has 'Properties.GroupName' set to 'default' and traffic rules set in 'Properties'.", [security_group_name]),
+		"searchLine": search_values.searchLine
 	}
 }
 
-check_group_name(resource) {
+CxPolicy[result] {						# standalone rules
+	resource := input.document[i].Resources[security_group_name]
 	resource.Type == "AWS::EC2::SecurityGroup"
-	security_group := resource.Properties
-	security_group.GroupName == "default"
-	check_rules(security_group)
+
+	resource.Properties.GroupName == "default"
+	rules := search_for_standalone_rules(security_group_name, input.document[i])
+	rule := rules.rule_list[x]
+
+	search_values := check_standalone_rule(security_group_name, rule, rules.names[x])
+
+	result := {
+		"documentId": input.document[i].id,
+		"resourceType": "AWS::EC2::SecurityGroup",
+		"resourceName": cf_lib.get_resource_name(resource, security_group_name),
+		"searchKey": search_values.searchKey,
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Any 'AWS::EC2::SecurityGroup' with 'Properties.GroupName' set to 'default' should not have any traffic rules set.",
+		"keyActualValue": sprintf("'Resources.%s' has 'Properties.GroupName' set to 'default' and a standalone '%s' rule set.", [security_group_name, rule.Type]),
+		"searchLine": search_values.searchLine
+	}
 }
 
-check_rules(security_group) {
-	count(security_group.SecurityGroupIngress) != 0
+search_for_standalone_rules(sec_group_name, doc) = rules_with_names {
+  rule_types := ["AWS::EC2::SecurityGroupIngress", "AWS::EC2::SecurityGroupEgress"]
+  resources := doc.Resources
+
+  names := [name |
+    rule := resources[name]
+    rule.Type == rule_types[_]
+    cf_lib.get_name(rule.Properties.GroupId) == sec_group_name
+  ]
+
+  rules_with_names := {
+    "rule_list": [resources[name] | name := names[_]],
+    "names": names
+  }
+} else = { "rule_list": [], "names": []}
+
+
+check_standalone_rule(security_group_name, rule, rule_name) = search_values {
```

Comment:
Since `check_standalone_rule` and `check_rules` are only used once, why are you returning the `searchKey` and `searchLine` like they are dynamic? If not justified, change it to be directly on the Policy result.